library/public/camera.h

```diff
@@ -0,0 +1,93 @@
+/**
+ * @class   camera
+ * @brief   Abstract class to control a camera in a window
+ *
+ * A class to control a camera in a window.
+ * There is in essence three API.
+ * The Position/FocalPoint/ViewUp API, the ViewMatrix API and the camera movements API.
+ * They can be mixed by using ViewMatrix API with the other APIs but it can result in unexpected
+ * results.
+ * When using the Position/FocalPoint/ViewUp API or the camera movements API,
+ * the view up is automatically orthogonalized.
+ */
+
+#ifndef f3d_camera_h
+#define f3d_camera_h
+
+#include "export.h"
+
+#include <array>
+#include <string>
+
+namespace f3d
+{
+class F3D_EXPORT camera
+{
+public:
+  typedef std::array<double, 3> vector3_t;
```

Comment:
You should consider making a distinction between points and vectors. They both are essentially a 3 value tuple, but they have different semantic meaning.

Also, if you implement Point3d and Vector3d classes, you can give them reasonable math operators like

```
Point + Vector => Point
Vector + Vector => Vector
Vector * Scalar => Vector
etc
```

which will make the library easier to work with.

Reply:
I agree. It's also convenient when transforming with a 4x4 matrix to know if the translation should be applied.

Reply:
I will add the types, not the operation, for now.

Reply:
Adressed in https://github.com/f3d-app/f3d/pull/355